python/tank/context.py

```diff
@@ -668,6 +668,83 @@ def create_empty(tk):
     """
     return Context(tk)
 
+def from_entities(tk, project, entity, step, task, user):
+    """
+    Construct a new context from the specified entities.
```

Comment:
Discussed the general approach offline with Manne as I shared some of his concerns about the initial implementation and the possibility of ending up with a context which not only isn't fully complete but may actually do damage if used in other parts of the system!

Instead I'm going to change this to a from_entity_dictionary() method that will make use of any linked entity information if available, specifically when creating a context from a task dictionary.  It will also try to fill out any missing entity information (e.g. name) by first looking in the path cache and then falling back to a sg query only if it needs to.